src/main/java/seedu/address/model/event/EventContainsKeyDateRangePredicate.java

```diff
@@ -31,8 +35,26 @@ public String getEndDate() {
      */
     @Override
     public boolean test(Event event) {
-        LocalDate date = event.getStartDate().getDate();
-        return !(date.isBefore(startDate) || date.isAfter(endDate));
+        LocalDate start = event.getStartDate().getDate();
+        LocalDate end = event.getEndDate().getDate();
+        List<LocalDate> eventDates = new ArrayList<>();
+        while (!start.isAfter(end)) {
+            eventDates.add(start);
+            start = start.plusDays(1);
+        }
+        LocalDate startRange = startDate;
+        LocalDate endRange = endDate;
+        List<LocalDate> dateRange = new ArrayList<>();
+        while (!startRange.isAfter(endRange)) {
+            dateRange.add(startRange);
+            startRange = startRange.plusDays(1);
```

Comment:
Looks like a very long piece of code that can be simplified. Perhaps you can look into my `EventDate#datesUntil` method that was previously created to return a Stream of EventDates.